sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_controlflow_pipeline.py

```diff
@@ -321,12 +321,42 @@ def test_pipeline(input_data, int_param, bool_param, float_param, str_param):
             },
         }
 
+    @pytest.mark.skip(reason="Backend not ready yet.")
+    def test_if_else_multiple_blocks(self, client: MLClient):
+        hello_world_component_no_paths = load_component(
+            source="./tests/test_configs/components/helloworld_component_no_paths.yml"
+        )
+        hello_world_component = load_component(source="./tests/test_configs/components/helloworld_component.yml")
+        basic_component = load_component(
+            source="./tests/test_configs/components/component_with_conditional_output/spec.yaml"
+        )
+
+        @pipeline(
+            compute="cpu-cluster",
+        )
+        def condition_pipeline():
+            result = basic_component()
 
-# @pytest.mark.skipif(
-#     condition=is_live(),
-#     # TODO: reopen live test when parallel_for deployed to canary
-#     reason="parallel_for is not available in canary."
-# )
+            node1 = hello_world_component_no_paths(component_in_number=1)
+
+            node2 = hello_world_component_no_paths(component_in_number=2)
+            node3 = hello_world_component(component_in_number=3, component_in_path=test_input)
+            node4 = hello_world_component(component_in_number=4, component_in_path=test_input)
+            condition(condition=result.outputs.output, false_block=[node1, node3], true_block=[node2, node4])
+
+        pipeline_job = condition_pipeline()
+        with include_private_preview_nodes_in_pipeline():
+            pipeline_job = assert_job_cancel(pipeline_job, client)
+
+        dsl_pipeline_job_dict = omit_with_wildcard(pipeline_job._to_rest_object().as_dict(), *omit_fields)
+        assert dsl_pipeline_job_dict["properties"]["jobs"] == {}
```

Comment:
a little bit confused: why is jobs empty here?

Reply:
The test not finished yet. Will update when backend supports this

Reply:
Updated